src/types.rs

```diff
@@ -208,13 +208,31 @@ pub struct Waypoint {
 
     // <magvar> degreesType </magvar> [0..1] ?
     // <geoidheight> xsd:decimal </geoidheight> [0..1] ?
-    // <fix> fixType </fix> [0..1] ?
-    // <sat> xsd:nonNegativeInteger </sat> [0..1] ?
-    // <hdop> xsd:decimal </hdop> [0..1] ?
-    // <vdop> xsd:decimal </vdop> [0..1] ?
-    // <pdop> xsd:decimal </pdop> [0..1] ?
-    // <ageofdgpsdata> xsd:decimal </ageofdgpsdata> [0..1] ?
-    // <dgpsid> dgpsStationType </dgpsid> [0..1] ?
+
+    /// Type of GPS fix. `none` means GPS had no fix. To signify "the fix info
+    /// is unknown", leave out `fix` entirely. Value comes from the list
+    /// `{'none'|'2d'|'3d'|'dgps'|'pps'}`, where `pps` means that the military
+    /// signal was used.
+    pub fix: Option<String>,
```

Comment:
What do you think about using an `enum` here? This way users wouldn't need to validate results because we'd already have validated them.

We could probably add a `fix.rs` and use it like [`fix::consume(reader)`](https://github.com/georust/rust-gpx/blob/c21aae75bd438954a30f7b61eac9fc87073de02d/src/parser/string.rs#L10) or something, thoughts?